columnflow/production/cms/dy.py

```diff
@@ -0,0 +1,459 @@
+# coding: utf-8
+
+"""
+Column production methods related to Drell-Yan reweighting.
+"""
+
+from __future__ import annotations
+
+import law
+
+from dataclasses import dataclass
+
+from columnflow.production import Producer, producer
+from columnflow.util import maybe_import 
+from law.util import InsertableDict
+from columnflow.columnar_util import set_ak_column
+
+np = maybe_import("numpy")
+ak = maybe_import("awkward")
+vector = maybe_import("vector")
+
+logger = law.logger.get_logger(__name__)
+
+
+@dataclass
+class DrellYanConfig:
+    era: str
+    order: str
+    correction: str
+    unc_correction: str
+
+    def __post_init__(self) -> None:
+        if (
+            not self.era or
+            not self.order or
+            not self.correction or
+            not self.unc_correction
+        ):
+            raise ValueError(
+                "Incomplete dy_weight_config: missing era, order, correction or unc_correction.",
+            )  # noqa
+
+
+@producer(
+    uses={"GenPart.*"},
+    produces={
+        "gen_dilepton_pt",
+        "gen_dilepton_vis.{pt,phi}",
+        "gen_dilepton_all.{pt,phi}",
```

Comment:
Just to visually simplify this a bit.

```suggestion
        "gen_dilepton_pt", "gen_dilepton_{vis,all}.{pt,phi}",
```